marker-api.js

```diff
@@ -1,780 +1,780 @@
+              message,
+              unique_img_hash;
+
+          // Get form data and unique hash
+          parse_marker_post(req, res).then(save_new_marker, handle_marker_data_failure);
+          return;
+      },
+
+      getMarkers: function (req, res) {
+          var async_operations = [],
+              marker_ids = [],
+              photo_requests = {},
+              returnData,
+              receivedData;
+
+          console.log('Get markers...');
+
+          receivedData = req.body.markers;
+          if (!receivedData || !receivedData.length) {
+             return res.status(422).json({message: 'Invalid Data passed with request'}); 
+          }
+
+          // Make two separate arrays for data retrieval and photo retrieval
+          receivedData.forEach((marker) => {
+
+              if (marker.public_id) {
+                  marker_ids.push(marker.public_id);
+
+                  // marker.photo_size is Array OR string
+                  if (marker.photo_size) {
+                      photo_requests[marker.public_id] = marker.photo_size;
+                  }
+              }
+          });
+
+          // No ids
+          if (!marker_ids.length) {
+             return res.status(422).json({message: 'Invalid Data passed with request'}); 
+          }
+
+          // Get marker data
+          async_operations.push(get_marker_data(marker_ids));
+
+          // Get image data (if requested)
+          if (Object.keys(photo_requests).length) {
+              async_operations.push(get_photo_b64(photo_requests));
+          }
+
+          Q.all(async_operations)
+              .spread(function (data, img_data) {
+
+                  returnData = data;
+
+                  // Insert img_data if returned
+                  returnData = returnData.map((marker) => {
+                      if (img_data[marker._id]) {
+                          marker.photos = {};
+                          
+                          // return data keyed by size
+                          for (var size in img_data[marker._id]) {
+                              marker.photos[size] = img_data[marker._id][size];
+                          }
+                          console.log('Including b64 img data: ' + Object.keys(marker.photos));
+                      }
+                      return marker;
+                  });
+
+                  res.status(200).json({
+                      message: 'found marker',
+                      data: returnData
+                  });
+                  
+              }).catch(function (reason) {
+                 res.status(reason.status).json({message: reason.message}); 
+              });
+
+      },
+
+      editMarker: function (req, res) {
+          var tags = req.body.tags,
+              marker_id = req.body.public_id;
+
+          // Only tags are editable for now
+          if (tags) {
+
+              // Update marker data
+              Marker.findOne({'_id': marker_id}, function (err, marker) {
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/charlieblevins/dukweb/pullRequest?prid=9134160)
```suggestion
              Marker.findOne({"_id": marker_id}, function (err, marker) {
```